e2e/rdctl.e2e.spec.ts

```diff
@@ -255,6 +255,92 @@ test.describe('Command server', () => {
+        const backupPath = path.join(paths.appHome, 'rd-engine.json.bak');
+        const mvCommand = os.platform() === 'win32' ? 'ren' : 'mv';
+
+        test.beforeAll(async() => {
+          const dataRaw = await fs.promises.readFile(configFilePath, 'utf-8');
+
+          serverState = JSON.parse(dataRaw);
+          parameters = [`--password=${ serverState.password }`,
+            `--port=${ serverState.port }`,
+            `--user=${ serverState.user }`,
+          ];
+          try {
+            await spawnFile(mvCommand, [configFilePath, backupPath]);
+          } catch (err) {
+            console.log(`Error trying to ${ mvCommand } ${ configFilePath } ${ backupPath }: `, err);
+            expect(err).toBeUndefined();
+          }
+        });
+        test.afterAll(async() => {
+          try {
+            await spawnFile(mvCommand, [backupPath, configFilePath]);
+          } catch (err) {
+            console.log(`Error trying to ${ mvCommand } ${ backupPath } ${ configFilePath }: `, err);
+            expect(err).toBeUndefined();
+          }
+        });
+
+        test('it complains with no parameters,', async() => {
+          const { stdout, stderr, error } = await rdctl(['list-settings']);
+
+          expect(error).toBeDefined();
+          expect(stderr).toContain(`Error: open ${ configFilePath }: no such file or directory`);
+          expect(stderr).not.toContain('Usage:');
+          expect(stdout).toEqual('');
+        });
+
+        test('it works with all parameters,', async() => {
+          const { stdout, stderr, error } = await rdctl(parameters.concat(['list-settings']));
+
+          expect(error).toBeUndefined();
+          expect(stderr).toEqual('');
+          expect(stdout).toMatch(/"kubernetes":/);
+          const settings = JSON.parse(stdout);
+
+          expect(['version', 'kubernetes', 'portForwarding', 'images', 'telemetry', 'updater', 'debug', 'pathManagementStrategy']).toMatchObject(Object.keys(settings));
+        });
+        test("it complains when some parameters aren't specified", async() => {
+          for (let idx = 0; idx < parameters.length; idx += 1) {
+            const partialParameters: string[] = parameters.slice(0, idx).concat(parameters.slice(idx + 1));
+            const { stdout, stderr, error } = await rdctl(partialParameters.concat(['list-settings']));
+
+            expect(error).toBeDefined();
+            expect(stderr).toContain(`Error: open ${ configFilePath }: no such file or directory`);
+            expect(stderr).not.toContain('Usage:');
+            expect(stdout).toEqual('');
+          }
+        });
+        test.describe('when a non-existent config file is specified', () => {
+          test('it fails even when all parameters are specified', async() => {
+            let badConfigFile = '/less/salt/more/gravy.json';
+            let i = 0;
+
+            // Ensure the specified configFile doesn't exist. Give up if we have to add 100 x's to the end.
```

Comment:
Why not just use [`fs.promises.mkdtemp()`](https://nodejs.org/docs/latest-v16.x/api/fs.html#fspromisesmkdtempprefix-options) to make a guaranteed non-existant directory (and therefore know that it doesn't contain any named config file)?

Reply:
Good point.